File: app/src/main/java/dev/ragnarok/fenrir/domain/ILikesInteractor.java
package dev.ragnarok.fenrir.domain;

import java.util.List;

import dev.ragnarok.fenrir.model.Owner;
import io.reactivex.rxjava3.core.Single;

public interface ILikesInteractor {
    String FILTER_LIKES = "likes";
    String FILTER_COPIES = "copies";

    Single<List<Owner>> getLikes(int accountId, String type, int ownerId, int itemId, String filter, int count, int offset);
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/domain/ILikesInteractor.kt
package dev.ragnarok.fenrir.domain

import dev.ragnarok.fenrir.model.Owner
import io.reactivex.rxjava3.core.Single

interface ILikesInteractor {
    fun getLikes(
        accountId: Long,
        type: String?,
        ownerId: Long,
        itemId: Int,
        filter: String?,
        count: Int,
        offset: Int
    ): Single<List<Owner>>

    companion object {
        const val FILTER_LIKES = "likes"
        const val FILTER_COPIES = "copies"
    }
}